src/Common/src/Microsoft/Win32/SafeHandles/GssSafeHandles.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.Win32.SafeHandles
+{
+    /// <summary>
+    /// Wrapper around a gss_name_t_desc*
+    /// </summary>
+    internal sealed class SafeGssNameHandle : SafeHandle
+    {
+        public static SafeGssNameHandle CreateUser(string name)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(name), "Invalid user name passed to SafeGssNameHandle create");
+            SafeGssNameHandle retHandle;
+            Interop.NetSecurityNative.Status minorStatus;
+            Interop.NetSecurityNative.Status status = Interop.NetSecurityNative.ImportUserName(
+                out minorStatus, name, Encoding.UTF8.GetByteCount(name), out retHandle);
```

Comment:
@stephentoub Looking at this line, I think we're looping the string at least 3 times:

1) The direct call to `UTF8.GetByteCount`
The the marshaller converts it to UTF8:
2) The internal sizing call to GetByteCount
3) Building the array.

For small strings this isn't a huge deal, but is this pattern our best practice?  Or should we just call UTF8.GetBytes() directly and pass the array and array length?

UCS-2 or ASCII are much easier, since they share a Length value with the string :).